### PR TITLE
Make the kafka streams extension to read the default kafka broker configuration

### LINKED ISSUE #15845: Kafka Streams extension does not read the "default broker configuration" map
## Describe the bug

The Kafka Streams extension does not read the configuration provided by the kafka-client extension.
The Kafka Streams extension used a different approach for configuration which won't work with service bindings as the support of service bindings is based on the "default configuration" approach (basically it produces a `Map<String, Object>` that any Kafka related application can read.

We need to extend the Kafka Streams configuration to read from the map. 
In `io.quarkus.kafka.streams.runtime.KafkaStreamsProducer#getStreamsProperties`, we could add a lookup for the default configuration and if there, pre-populate the Kafka Streams configuration.

### PR DESCRIPTION
**Pull request by cescoffier**
Fix #15845 

As a consequence, it will read the configuration produced using service bindings.

### CODE REVIEW

File: integration-tests/kafka-streams/src/main/resources/application.properties [hunk 1]
```diff
@@ -8,10 +8,11 @@ quarkus.kafka-streams.topics=streams-test-categories,streams-test-customers
 quarkus.kafka-streams.schema-registry-key=apicurio.registry.url
 quarkus.kafka-streams.schema-registry-url=http://localhost:8080
 
-quarkus.kafka-streams.security.protocol=SSL
-quarkus.kafka-streams.ssl.truststore.location=./target/classes/ks-truststore.p12
-quarkus.kafka-streams.ssl.truststore.password=Z_pkTh9xgZovK4t34cGB2o6afT4zZg0L
-quarkus.kafka-streams.ssl.truststore.type=PKCS12
```
Comment (geoand) [inline integration-tests/kafka-streams/src/main/resources/application.properties:14]: Is this a breaking change? It looks like it

Comment (geoand) [reply integration-tests/kafka-streams/src/main/resources/application.properties:14]: Or does it work in addition?

Comment (cescoffier) [reply integration-tests/kafka-streams/src/main/resources/application.properties:14]: Now it will support both.

Comment (cescoffier) [reply integration-tests/kafka-streams/src/main/resources/application.properties:14]: I can actually mix and match the properties

Comment (geoand) [reply integration-tests/kafka-streams/src/main/resources/application.properties:14]: We should document that somewhere - and which takes precedence when they both exist

**Review by geoand — APPROVED** [review]

**Review by machi1990 — APPROVED** [review]

**Pull request merged**